chore(ssm): update simple name description and documentation (#30653)
aws/aws-cdk: packages/aws-cdk-lib/aws-ssm/README.md

are validated against the pattern and an exception is raised if a value
provided does not comply.

## Using Tokens in parameter name

When using [CDK Tokens](https://docs.aws.amazon.com/cdk/v2/guide/tokens.html) in parameter name,

> you need to explicitly set `simpleName` property. Setting `simpleName` to incorrect boolean
> value may result in unexpected behaviours, such as getting duplicate '/' in the parameter ARN
> or missing '/' in the parameter ARN.

Reviewer: ```suggestion
you need to explicitly set the `simpleName` property. Setting `simpleName` to an incorrect boolean
value may result in unexpected behaviours such as having a duplicate '/' in the parameter ARN
or missing a '/' in the parameter ARN.
```

`simpleName` is used to indicates whether the parameter name is a simple name. A parameter name
without any '/' is considered a simple name, thus you should set `simpleName` to `true`.
If the parameter name includes '/', set `simpleName` to `false`.

```ts
import * as lambda from 'aws-cdk-lib/aws-lambda';

const simpleParameter = new ssm.StringParameter(this, 'StringParameter', {

>   // the parameter name is simple

Reviewer: ```suggestion
  // the parameter name does not contain any '/'
```

  parameterName: 'parameter',
  stringValue: 'SOME_VALUE',
  simpleName: true, // set `simpleName` to true
});

declare const func: lambda.IFunction;
const nonSimpleParameter = new ssm.StringParameter(this, 'StringParameter', {

>   // the parameter name is not simple

Reviewer: ```suggestion
  // the parameter name contains '/'
```
